x-pack/platform/plugins/shared/streams/server/lib/streams/stream_crud.ts

```diff
@@ -274,3 +282,145 @@ export async function getDataStream({
   }
   return dataStream;
 }
+
+// In case this retention is not present in cluster.
+// This is extracted from the docs that indicate that a thirty day (30d) retention is applied to failure store data:
+// https://www.elastic.co/docs/manage-data/data-store/data-streams/failure-store#manage-failure-store-lifecycle
+const DEFAULT_RETENTION_PERIOD = '30d';
+
+export async function getDefaultRetentionValue({
+  scopedClusterClient,
+}: {
+  scopedClusterClient: IScopedClusterClient;
+}): Promise<string> {
+  const { persistent, defaults } =
+    await scopedClusterClient.asSecondaryAuthUser.cluster.getSettings({ include_defaults: true });
+  const persistentDSRetention = persistent?.data_streams?.lifecycle?.retention?.failures_default;
+  const defaultsDSRetention = defaults?.data_streams?.lifecycle?.retention?.failures_default;
+  return persistentDSRetention ?? defaultsDSRetention ?? DEFAULT_RETENTION_PERIOD;
+}
+
+export async function getFailureStore({
+  name,
+  scopedClusterClient,
+}: {
+  name: string;
+  scopedClusterClient: IScopedClusterClient;
+}): Promise<FailureStore> {
+  const dataStream = await getDataStream({ name, scopedClusterClient });
+  const defaultRetentionPeriod = await getDefaultRetentionValue({ scopedClusterClient });
+
+  return {
+    enabled: !!dataStream.failure_store?.enabled,
+    retentionPeriod: {
+      // @ts-expect-error
```

Comment:
can we instead type the response to include lifecycle and leave a TODO to remove it once the client is updated ?